templates/common/_base/files/configure-ovs-network.yaml

```diff
@@ -490,6 +490,63 @@ contents:
       echo ""
     }
 
+    get_ip_from_ip_hint_file() {
+      local ip_hint_file="$1"
+      if [[ ! -f "${ip_hint_file}" ]]; then
+        return
+      fi
+      ip_hint=$(cat "${ip_hint_file}")
+      echo ${ip_hint}
+    }
+
+    try_to_bind_address() {
+      local ip_hint_file="$1"
+      ip_hint=$(get_ip_from_ip_hint_file "${ip_hint_file}")
+      if [[ -z "${ip_hint}" ]]; then
+        return
+      fi
+      while true; do
+        echo "Trying to bind ${ip_hint}"
+        exit_code=$(timeout 1s nc -l ${ip_hint}; echo $?)
+        if [[ exit_code -eq 124 ]]; then
+          echo "Address bound successfully"
+          break
+        fi
+        sleep 1
+      done
+    }
+
+    get_interface_from_ip_hint() {
+      local ip_hint=""
+      local ip_hint_file="$1"
+      local extra_bridge="$2"
+      local iface=""
+
+      ip_hint=$(get_ip_from_ip_hint_file "${ip_hint_file}")
+      if [[ -z "${ip_hint}"  ]]; then
+         return
+      fi
+      # check ipv4
+      if [[ -n "${extra_bridge}" ]]; then
+        iface=$(ip route show | grep -v "br-ex1" | grep -v "${extra_bridge}" | grep -i "${ip_hint}" | awk '{ if ($4 == "dev") { print $5; exit } else if ($2 == "dev") { print $3; exit } }')
```

Comment:
This is finicky. I would extract this to it's own function and pipe `ip -j` output to `jq` to find the interface with given address?

Example:
```
ip -j addr | jq ".[] | select(any(.addr_info[]; .local==\"fe80::1\" and .family==\"inet6\")) | .ifname"
```

I am confused though, I am understanding the contents of the node ip hint file is the ip address of an interface. Is this correct? If not, could  you add a comment explaining what that is?

Reply:
Yep, it is actual ip that node ip chooses. This function should find interface matching this ip

Reply:
sounds reasonable to use jq i just stick to the same code logic as it was before. Will change

Reply:
Yeah, we did not have `jq` available before but since we have it we are resorting to it rather than parsing the normal ip output.